src/signature.rs

```diff
@@ -302,14 +302,15 @@ pub use ec::curve25519::ed25519::{
     ED25519,
 
     Ed25519KeyPair,
+    Ed25519KeyPairComponents,
     ED25519_PKCS8_V2_LEN,
     ED25519_PUBLIC_KEY_LEN,
 };
 
 pub use pkcs8::PKCS8Document;
 
 #[cfg(all(feature = "rsa_signing", feature = "use_heap"))]
-pub use rsa::signing::{RSAKeyPair, RSASigningState};
+pub use rsa::signing::{RSAKeyPair, RSAKeyPairComponents, RSASigningState};
```

Comment:
Ditto; this should be exposed from `ring::signature::primitives`, not `ring::signature`.